packages/editor/src/client/routes/commentList.tsx

```diff
@@ -0,0 +1,408 @@
+            <DescriptionListItem label={t('comments.panels.revisions')}>
+              <Timeline align="left">
+                {currentComment?.revisions?.length
+                  ? currentComment?.revisions?.map(({text, createdAt}, i) => (
+                      <Timeline.Item key={i}>
+                        <div>{new Date(createdAt).toLocaleString()}</div>
+                        <RichTextBlock
+                          disabled
+                          // TODO: remove this
+                          onChange={console.log}
+                          value={text}
+                        />
+                      </Timeline.Item>
+                    ))
+                  : null}
+              </Timeline>
+            </DescriptionListItem>
+            {confirmAction === ConfirmAction.Reject ||
+            confirmAction === ConfirmAction.RequestChanges ? (
+              <DescriptionListItem
+                label={t(
+                  confirmAction === ConfirmAction.Reject
+                    ? 'comments.panels.rejectionReason'
+                    : 'comments.panels.requestChangesReason'
+                )}>
+                <Dropdown
+                  title={t(
+                    confirmAction === ConfirmAction.Reject
+                      ? 'comments.panels.rejectionReason'
+                      : 'comments.panels.requestChangesReason'
+                  )}
+                  placement="topStart">
+                  <Dropdown.Item
+                    key={CommentRejectionReason.Spam}
+                    active={CommentRejectionReason.Spam === rejectionReason}
+                    onSelect={() => setRejectionReason(CommentRejectionReason.Spam)}>
+                    {CommentRejectionReason.Spam}
+                  </Dropdown.Item>
+                  <Dropdown.Item
+                    key={CommentRejectionReason.Misconduct}
+                    active={CommentRejectionReason.Misconduct === rejectionReason}
+                    onSelect={() => setRejectionReason(CommentRejectionReason.Misconduct)}>
+                    {CommentRejectionReason.Misconduct}
+                  </Dropdown.Item>
+                </Dropdown>
+              </DescriptionListItem>
+            ) : null}
+          </DescriptionList>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            color={mapCommentActionToColor(confirmAction)}
+            disabled={isApproving || isRequestingChanges || isRejecting}
+            onClick={async () => {
+              if (!currentComment) return
+              switch (confirmAction) {
+                case ConfirmAction.Approve:
+                  await approveComment({
+                    variables: {
+                      id: currentComment.id
+                    }
+                  })
+                  setConfirmationDialogOpen(false)
+                  break
+                case ConfirmAction.RequestChanges:
+                  if (!rejectionReason) {
```

Comment:
Disable the confirm button if there is no rejectionReason. It's bad UX if you allow an action to happen if the data is not complete.